core/src/main/java/org/wso2/msf4j/AbstractSessionManager.java

```diff
@@ -48,55 +47,70 @@
      */
     private static final int SESSION_ID_LENGTH = 16;
 
-    private Map<String, Session> sessions = new ConcurrentHashMap<>();
     private SessionIdGenerator sessionIdGenerator = new SessionIdGenerator();
 
     private ScheduledExecutorService sessionExpiryChecker;
 
+    private final MicroservicesRegistry microservicesRegistry;
+
+    public AbstractSessionManager(MicroservicesRegistry microservicesRegistry) {
+        this.microservicesRegistry = microservicesRegistry;
+    }
+
+    @Override
     public final void init() {
         sessionIdGenerator.setSessionIdLength(SESSION_ID_LENGTH);
-        loadSessions(sessions);
+        Map<String, Map<String, Session>> sessions = loadSessions();
 
         // Session expiry scheduled task
         sessionExpiryChecker = Executors.newScheduledThreadPool(1);
         sessionExpiryChecker.scheduleAtFixedRate(() ->
-                        sessions.values().parallelStream()
-                                .filter(session ->
-                                        (System.currentTimeMillis() - session.getLastAccessedTime() >=
-                                                session.getMaxInactiveInterval() * 60 * 1000))
-                                .forEach(Session::invalidate),
-                30, 30, TimeUnit.SECONDS);
+                sessions.values().forEach(entry -> entry.values().parallelStream()
```

Comment:
Wouldn't this create a NPE in some cases, like when using DefaultSessionManager?